android/java/org/chromium/chrome/browser/rewards/adaptive_captcha/AdaptiveCaptchaHelper.java

```diff
@@ -0,0 +1,275 @@
+        integrityTokenResponseTask.addOnSuccessListener(
+                response -> { attestPaymentId(captchaId, paymentId, response.token(), nonce); });
+        integrityTokenResponseTask.addOnFailureListener(e -> {
+            Log.e(TAG, "addOnFailureListener : " + e.getMessage());
+            recordFailureAttempt();
+        });
+    }
+
+    private static void attestPaymentId(
+            String captchaId, String paymentId, String integrityToken, String uniqueValue) {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(() -> {
+            HttpURLConnection urlConnection = null;
+            String attestPaymentIdUrl =
+                    BraveRewardsNativeWorker.getInstance().getAttestationURLWithPaymentId(
+                            paymentId);
+            Log.e(TAG, attestPaymentIdUrl);
+            NetworkTrafficAnnotationTag annotation = NetworkTrafficAnnotationTag.createComplete(
+                    "Brave attestation api android",
+                    "semantics {"
+                            + "  sender: 'Brave Android app'"
+                            + "  description: "
+                            + "    'This api attests play integrity token for the given payment ID'"
+                            + "  trigger: 'When payment id as captcha scheduled with integrity token'"
+                            + "  data:"
+                            + "    'integrity token, unique value, package name'"
+                            + "  destination: Brave grant endpoint"
+                            + "}"
+                            + "policy {"
+                            + "  cookies_allowed: NO"
+                            + "  policy_exception_justification: 'Not implemented.'"
+                            + "}");
+            String logMessage = "";
+            try {
+                URL url = new URL(attestPaymentIdUrl);
+                urlConnection =
+                        (HttpURLConnection) ChromiumNetworkAdapter.openConnection(url, annotation);
+                urlConnection.setDoOutput(true);
+                urlConnection.setRequestMethod("PUT");
+                urlConnection.setUseCaches(false);
+                urlConnection.setRequestProperty("Content-Type", "application/json");
+                urlConnection.connect();
+
+                JSONObject jsonParam = new JSONObject();
+                jsonParam.put("integrityToken", integrityToken);
+                jsonParam.put("uniqueValue", uniqueValue);
+                jsonParam.put("packageName", ContextUtils.getApplicationContext().getPackageName());
+
+                OutputStream outputStream = urlConnection.getOutputStream();
+                byte[] input = jsonParam.toString().getBytes(StandardCharsets.UTF_8.name());
+                outputStream.write(input, 0, input.length);
+                outputStream.flush();
+                outputStream.close();
```

Comment:
It seems like several of the methods in this class share a lot of boilerplate networking code. Maybe it can be extracted into a helper?